src/workerd/api/sockets.c++

```diff
@@ -263,14 +263,30 @@ jsg::Ref<Socket> connectImpl(
 }
 
 jsg::Promise<void> Socket::close(jsg::Lock& js) {
-  // Forcibly close the readable/writable streams.
-  auto cancelPromise = readable->getController().cancel(js, kj::none);
-  auto abortPromise = writable->getController().abort(js, kj::none);
-  // The below is effectively `Promise.all(cancelPromise, abortPromise)`
-  return cancelPromise.then(js, [abortPromise = kj::mv(abortPromise), this](jsg::Lock& js) mutable {
-    return abortPromise.then(js, [this](jsg::Lock& js) {
-      resolveFulfiller(js, kj::none);
-      return js.resolvedPromise();
+  if (isClosing) {
+    return closedPromiseCopy.whenResolved(js);
```

Comment:
Nit; If we can't get rid of these `...Copy` promises, is there a more descriptive name we could use? If that's too hard though, let it be.

Reply:
I think "copy" describes the intent much better.

Reply:
Alright then!